lib/jxl/modular/encoding/encoding.cc

```diff
@@ -358,6 +358,30 @@ Status DecodeModularChannelMAANS(BitReader *br, ANSSymbolReader *reader,
 
 GroupHeader::GroupHeader() { Bundle::Init(this); }
 
+Status ValidateChannelDimensions(const Image &image,
+                                 const ModularOptions &options) {
+  size_t nb_channels = image.channel.size();
+  for (bool is_dc : {true, false}) {
+    size_t group_dim = options.group_dim * (is_dc ? kBlockDim : 1);
+    size_t c = image.nb_meta_channels;
+    for (; c < nb_channels; c++) {
+      const Channel& ch = image.channel[c];
+      if (ch.w > group_dim || ch.h > group_dim) break;
```

Comment:
This `group_dim` should be `options.group_dim`, also in the DC case, unless I'm missing something.

Reply:
Right! Fixed.